### PR TITLE
Tftpboot

### PR DESCRIPTION
**Pull request by mikajuvonen67**
Hi, first version. I greatefully accpet any comments.

### CODE REVIEW

File: board/demo/configuration/post-build.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+echo "NAME=Configuration"                               >$TARGET_DIR/etc/os-release
+echo "VERSION=1.0.0"                              >>$TARGET_DIR/etc/os-release
+echo "ID=configuration"                                >>$TARGET_DIR/etc/os-release
+echo "VERSION_ID=1.0.0"                           >>$TARGET_DIR/etc/os-release
+echo "PRETTY_NAME=\"Configuration Demo\""              >>$TARGET_DIR/etc/os-release
+
+# Provide symbolic names for routes added by the configuration daemon
+if ! grep -q configuration $TARGET_DIR/etc/iproute2/rt_protos; then
+    echo "254\tconfiguration" >>$TARGET_DIR/etc/iproute2/rt_protos
+fi
```
Comment (wkz) [inline board/demo/configuration/post-build.sh]: Is this used? Is the app installing routes with a custom protocol?

File: package/configuration/Config.in [hunk 2]
```diff
@@ -0,0 +1,10 @@
+config BR2_PACKAGE_CONFIGURATION
+	bool "configuration"
+	select BR2_PACKAGE_JANSSON
+	select BR2_PACKAGE_LIBEV
+	select BR2_PACKAGE_LIBTEAM
+	select BR2_PACKAGE_NFTABLES
+	select BR2_PACKAGE_SMCROUTE
+	select BR2_PACKAGE_IPROUTE2
+	help
+	  Fill in some help text here.
```
Comment (wkz) [inline package/configuration/Config.in]: This also looks like copypasta from backbone.

Can we limit the selection? We don't want to create folklore by tricking people into thinking that these packages are always needed.

File: src/configuration/README.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
+Download WeOS cfg from configuration application
+=====================================================
+
+Introduktion
```
Comment (wkz) [inline src/configuration/README.md]:
```suggestion
Introduction
```

File: board/demo/configuration/post-build.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+TARGET_DIR=$1
+
+# This is a symlink to /usr/lib/os-release, so we remove this to keep
+# original Buildroot information.
+rm $TARGET_DIR/etc/os-release
+
+echo "NAME=Configuration"                               >$TARGET_DIR/etc/os-release
```
Comment (wkz) [inline board/demo/configuration/post-build.sh]: "configuration" is very nondescript. How about "dhcp-boot" or "tftp-boot"?

File: package/configuration/configuration.mk [hunk 5]
```diff
@@ -0,0 +1,30 @@
+################################################################################
+#
+# configuration
+#
+################################################################################
+
+CONFIGURATION_VERSION = 1
+CONFIGURATION_LICENSE = MIT
+CONFIGURATION_DEPENDENCIES = iproute2 jansson libev libteam nftables smcroute
```
Comment (wkz) [inline package/configuration/configuration.mk]: Actual dependencies only please.

File: src/configuration/README.md [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+-------------
+
+The `configuration` application is an example of how a container application could
+create or alter a configuration file. In a real world scenario a non nativ configuration file
+could be converted to OS nativ format.
```
Comment (wkz) [inline src/configuration/README.md]:
```suggestion
could be converted to OS native format. 
```
Ignoring spelling from now on, please go through the whole doc with `ispell` or similar.

File: src/configuration/cfg-provider.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# Set up the default interface
+ip link set dev eth0 up
+ip addr add 10.0.0.89/24 dev eth0
```
Comment (wkz) [inline src/configuration/cfg-provider.sh]:
```suggestion
ip addr add 10.0.0.1/24 dev eth0
```
Any reason we cannot go with the first address in the subnet? Perhaps we should even limit the network to a `/30` to make it clear that this is, in effect, a point-to-point connection?

File: src/configuration/cfg-provider.sh [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# Set up the default interface
+ip link set dev eth0 up
```
Comment (wkz) [inline src/configuration/cfg-provider.sh]: Can we not setup the network in `/etc/network/interfaces` as usual?

File: src/configuration/configuration.c [hunk 9]
```diff
@@ -0,0 +1,186 @@
+#include <ev.h>
```
Comment (wkz) [inline src/configuration/configuration.c]: Could we not just have this be a regular UNIX-style program that processes the file and then exits? Then we run that as a task in `/etc/inittab` (I suppose there is no time to move to a NetBox base using finit?) before starting dnsmasq as a regular service.

We could even replace the program with a direct call to `sed`.

Comment (troglobit) [reply src/configuration/configuration.c]: +1 that would be great.

**Comment by troglobit** [comment]
Great work so far!  I concur fully with Tobias' review.

> Hint: to resolve each you can create a separate commit for each highlighted issue on the same branch as this pull request is from, and when you push the changes, GitHub will mark each as resolved automatically.

File: board/demo/configuration/post-build.sh (hunk 1 above)
Comment (mikajuvonen67) [reply board/demo/configuration/post-build.sh]: Removed

File: package/configuration/Config.in (hunk 2 above)
Comment (mikajuvonen67) [reply package/configuration/Config.in]: Removed unnecessary packages

File: src/configuration/README.md (hunk 3 above)
Comment (mikajuvonen67) [reply src/configuration/README.md]: Rewritten and checked spelling.

File: board/demo/configuration/post-build.sh (hunk 4 above)
Comment (mikajuvonen67) [reply board/demo/configuration/post-build.sh]: Renamed application to dhcp-boot.

File: package/configuration/configuration.mk (hunk 5 above)
Comment (mikajuvonen67) [reply package/configuration/configuration.mk]: Removed unnecessary dependencies.

File: src/configuration/cfg-provider.sh (hunk 8 above)
Comment (mikajuvonen67) [reply src/configuration/cfg-provider.sh]: Done

File: src/configuration/configuration.c (hunk 9 above)
Comment (mikajuvonen67) [reply src/configuration/configuration.c]: Changed to run a script that calls sed from /etc/inittab.

**Pull request merged**